src/Collapse.js

```diff
@@ -213,14 +235,14 @@ class Collapse extends React.Component {
               className,
               children.props.className,
               collapseStyles[state],
-              this.getDimension() === 'width' && 'width',
+              getDimension(dimension) === 'width' && 'width',
```

Comment:
**suggestion:** this isn't really relevant as set up any more; just pre-calculate the dimension once and for all up front in the function, and close over it in the callback handlers

Reply:
Yes ! I'm on it :)